doc/callbacks.rst

```diff
@@ -388,3 +388,12 @@ In the above examples the callback function is defined in the same scope as the
     solveProblem()
 
 This code can also be found in ``/JuMP/examples/simplelazy2.jl``.
+
+Exiting a callback early
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need to exit the optimization process earlier than a solver otherwise would, it is possible to throw a ``CallbackAbort`` exception in callback code::
+
+    throw(CallbackAbort())
+
+JuMP will take this exception and pass the appropriate error code to the solver so that the environment can be cleaned up. Note that the solver level will most likely throw an exception of its own, e.g. Gurobi will throw a ``GurobiError``.
```

Comment:
You could argue that the solver should just return ``UserLimit`` instead of throwing an exception.